test/README.md

```diff
@@ -0,0 +1,22 @@
+Scylla in-source tests.
+
+For details on how to run the tests, see docs/dev/testing.md
+
+Shared C++ utils, libraries are in lib/, for Pythoin - pylib/
+
+alternator - client/server tests for Scylla Alternator
+unit, boost, raft - unit tests in C++
+cql-pytest - Python tests which connect to a single server and use CQL
+topology* - tests that set up clusters and add/remove nodes
+cql - approval tests that use CQL and pre-recorded output
+rest\_api - tests for Scylla REST API Port 9000
+
+If you can use an existing folder, consider adding your test to it.
+New folders should be used for new large categories/subsystems, 
+or when the test environment is significantly different from some existing
+suite, e.g. you plan to start scylladb with different configuration,
```

Comment:
I think this "different configuration" is a mistake, and I don't want to make it an "official" decision in this file. 
Consider dtest - we don't have 10 different dtest projects just because a few tests start Scylla with different configuration.

Reply:
dtest doesn't pool clusters, test.py does. clusters with the same configuration can be pooled. if you don't want your clusters to be pooled, don't create a separate test suite. No policy can replace common sense.

Reply:
I'm sorry, but I feel that you are saying one thing about "new folders", and doing another thing.
1. Recently you added at least two test/ subdirectories which do NOT pool clusters - each test creates its own cluster and then deletes it.
2. Pooling has nothing to do with "New folders should be used for new large categories/subsystems". Moreover, I don't know what this statement refers to - we don't have a separate test **directory** for CDC, for MV, or many other large subsystems. You're trying to enforce here a convention we didn't agree on yet, and we're not using yet.
3. In the long run, test.py does not need all the test that can reuse the same cluster configuration to be in a separate directory - it can also be a separate file, or maybe even deteremined automatically. Like you don't need to put code that malloc()s different sizes in different source files ;-)

Reply:
1) I think we will eventually get a dir for mv, cdc, etc. 
2) There may be more than one reason to create a directory and not to create one. This is not a simple decision
3) Please feel free to suggest your own text.

Reply:
I said this, and I'll say it again: No, I think it will be a mistake to get a dir for mv, cdc, etc.
Consider, for example, your favorite test suite: "cql", with text-only CQL tests.
Will you want to split it into separate subdirectories cql-mv, cql-cdc, etc.? And then also have boost-mv, boost-cdc, etc.? This makes no sense. Also, what about a test that checks both mv and cdc together, where do we put it?
None of this makes sense.
It makes sense to separate tests according to topic *loosely* into individual files (e.g., test/cql-pytest/test_materialized_view.py focuses on materialized views) or even subdirectories, but not into rigid top-level subdirectories - those will simply become a complete mess, and pretty soon you'll have dozens of them.

Reply:
https://github.com/tarantool/tarantool/tree/master/test
https://github.com/mysql/mysql-server/tree/trunk/mysql-test/suite

Something to explore as an inspirations. In MySQK, also a nice thing "collections" - i.e. you have directories, and you have collections, they are orthogonal to each other, something like our "modes", but not quite.

Reply:
I don't know why dozens of directories are a mess. Usually bundling everything together is more messy :), but I guess it's a matter of taste. What matters is if you can easily find a place for a new test and easily find who owns what among existing tests.